packages/astro/src/assets/vite-plugin-assets.ts

```diff
@@ -223,22 +226,31 @@ export default function assets({ settings }: { settings: AstroSettings }): vite.
 					if (settings.config.experimental.svg && /\.svg$/.test(id)) {
 						const { contents, ...metadata } = imageMetadata;
 						// We know that the contents are present, as we only emit this property for SVG files
-						return makeSvgComponent(metadata, contents!, {
-							mode: settings.config.experimental.svg.mode,
-						});
+						return {
```

Comment:
Makes it an object that returns `code` (rollup/rolldown compatible shape) and `moduleType` (only rolldown I think)

Reply:
What I find weird is that we return strings in all our plugins, however we must return an object only here. It doesn't make much sense. I mean, it does but we should add a comment at this point.

Reply:
i think in some cases we return objects. But returning a string or `{ code: "..." }` is the same, it just allows to add more options as documented at https://rollupjs.org/plugin-development/#load (source descriptor). And it will allow us to easily adopt rolldown module types (https://rolldown.rs/guide/in-depth/module-types).